crates/c/src/lib.rs

```diff
@@ -650,22 +650,21 @@ impl C {
         resolve: &Resolve,
         handle: &Handle,
         handles: &HashMap<&Handle, TypeId>,
+        owner: TypeOwner,
     ) -> Option<TypeId> {
-        let mut target = match handle {
-            Handle::Borrow(target) => target,
-            Handle::Own(target) => target,
-        };
+        let (Handle::Borrow(mut target) | Handle::Own(mut target)) = handle;
 
         loop {
-            if let TypeDefKind::Type(Type::Id(id)) = &resolve.types[*target].kind {
+            let def = &resolve.types[target];
+            if let TypeDefKind::Type(Type::Id(id)) = &def.kind {
                 let target_handle = match handle {
                     Handle::Borrow(_) => Handle::Borrow(*id),
                     Handle::Own(_) => Handle::Own(*id),
                 };
-                if let Some(ty) = handles.get(&target_handle) {
+                if let (true, Some(ty)) = (def.owner == owner, handles.get(&target_handle)) {
```

Comment:
The basic idea here is that we shouldn't generate `typedef` aliases for handles that alias across interfaces, since the target interface might not have been imported at all.  We might want to expand this to check if the target interface _is_ imported and do something different.  Or maybe take a step back and consider a broader re-design.

Reply:
Actually, I just reverted this part and the test still passes, so nevermind :).  I'll push an update.